[CI] Assertion fails in ReadOnlyEngineTests.testForceMergeOnReadOnlyEngine

**Build scan**: https://gradle-enterprise.elastic.co/s/7asjwbf6fix24

**Repro line**:

```
./gradlew ':server:test' --tests "org.elasticsearch.index.engine.ReadOnlyEngineTests.testForceMergeOnReadOnlyEngine" -Dtests.seed=7D74F9F1EDD7D72 -Dtests.security.manager=true -Dtests.locale=en-US -Dtests.timezone=Asia/Calcutta -Druntime.java=11
```

**Reproduces locally?**: No

**Applicable branches**: master

**Failure history**: 

Another failure happened on the 30th of December.

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testForceMergeOnReadOnlyEngine),sort:!(process.time-start,desc))

**Failure excerpt**:

```
java.lang.AssertionError: (No message provided)

at __randomizedtesting.SeedInfo.seed([7D74F9F1EDD7D72:9D5314158B1642B0]:0) |  
-- | --
  |   | at org.junit.Assert.fail(Assert.java:86) |  
  |   | at org.junit.Assert.assertTrue(Assert.java:41) |  
  |   | at org.junit.Assert.assertTrue(Assert.java:52) |  
  |   | at org.elasticsearch.index.engine.ReadOnlyEngineTests.testForceMergeOnReadOnlyEngine(ReadOnlyEngineTests.java:210)
```

--- a/server/src/test/java/org/elasticsearch/index/engine/ReadOnlyEngineTests.java
+++ b/server/src/test/java/org/elasticsearch/index/engine/ReadOnlyEngineTests.java
@@ -189,7 +189,6 @@ public class ReadOnlyEngineTests extends EngineTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/67232")
     public void testForceMergeOnReadOnlyEngine() throws IOException {
         IOUtils.close(engine, store);
         final AtomicLong globalCheckpoint = new AtomicLong(SequenceNumbers.NO_OPS_PERFORMED);
@@ -208,15 +207,16 @@ public class ReadOnlyEngineTests extends EngineTestCase {
                 engine.syncTranslog();
                 engine.flushAndClose();
                 numSegments = engine.getLastCommittedSegmentInfos().size();
-                assertTrue( numSegments > 1);
             }
 
             try (ReadOnlyEngine readOnlyEngine = new ReadOnlyEngine(config, null , null, true, Function.identity(), true)) {
-                UnsupportedOperationException exception = expectThrows(UnsupportedOperationException.class,
-                    () -> readOnlyEngine.forceMerge(
-                        true, numSegments-1, false, false, false, UUIDs.randomBase64UUID()));
-                assertThat(exception.getMessage(), equalTo("force merge is not supported on a read-only engine, " +
-                    "target max number of segments[" + (numSegments-1) + "], current number of segments[" + numSegments + "]."));
+                if (numSegments > 1) {
+                    final int target = between(1, numSegments - 1);
+                    UnsupportedOperationException exception = expectThrows(UnsupportedOperationException.class,
+                            () -> readOnlyEngine.forceMerge(true, target, false, false, false, UUIDs.randomBase64UUID()));
+                    assertThat(exception.getMessage(), equalTo("force merge is not supported on a read-only engine, " +
+                            "target max number of segments[" + target + "], current number of segments[" + numSegments + "]."));
+                }
 
                 readOnlyEngine.forceMerge(true, ForceMergeRequest.Defaults.MAX_NUM_SEGMENTS,
                     false, false, false, UUIDs.randomBase64UUID());

